prow/config/config_test.go

```diff
@@ -29,6 +29,77 @@ import (
 	"k8s.io/test-infra/prow/kube"
 )
 
+func TestSpyglassConfig(t *testing.T) {
+	testCases := []struct {
+		name            string
+		spyglassConfig  string
+		expectedViewers map[string][]string
+		expectError     bool
+	}{
+		{
+			name: "Default: build log, metadata, junit",
+			spyglassConfig: `
+spyglass:
+  viewers:
+    "started.json|finished.json": ["MetadataViewer"]
+    "build-log.txt": ["BuildLogViewer"]
+    "artifacts/junit.*\\.xml": ["JUnitViewer"]
+`,
+			expectedViewers: map[string][]string{
+				"started.json|finished.json": {"MetadataViewer"},
+				"build-log.txt":              {"BuildLogViewer"},
+				"artifacts/junit.*\\.xml":    {"JUnitViewer"},
+			},
+			expectError: false,
+		},
+	}
+	for _, tc := range testCases {
+		// save the config
+		spyglassConfigDir, err := ioutil.TempDir("", "spyglassConfig")
```

Comment:
This unit test format was taken from another in `config_test.go`, if it's a problem for you here it's probably also an issue there so this might be better handled in a separate PR

Reply:
Let's try to fix this here and do cleanup later in a follow-up. No use to adding more unit tests that need the filesystem.